src/Ocelot/Authorization/ScopesAuthorizer.cs

```diff
@@ -21,21 +16,32 @@ public Response<bool> Authorize(ClaimsPrincipal claimsPrincipal, List<string> ro
                 return new OkResponse<bool>(true);
             }
 
-            var values = _claimsParser.GetValuesByClaimType(claimsPrincipal.Claims, Scope);
+            var userScopes =
+                claimsPrincipal.Claims
+                    .Where(x => x.Type == ScopeClaimKey)
+                    .Select(x => x.Value)
+                    .ToArray();
 
-            if (values.IsError)
+            if (userScopes.Length == 1)
             {
-                return new ErrorResponse<bool>(values.Errors);
+                var userScope = userScopes[0];
+
+                var hasMultipleValues = userScope.Contains(" ");
+
+                if (hasMultipleValues)
+                {
+                    userScopes = userScope.Split(" ", StringSplitOptions.RemoveEmptyEntries);
+                }
+                else
+                {
+                    userScopes = new[] { userScope };
+                }
             }
 
-            var userScopes = values.Data;
-
-            var matchesScopes = routeAllowedScopes.Intersect(userScopes);
-
-            if (!matchesScopes.Any())
+            if (routeAllowedScopes.Any(s => !userScopes.Contains(s)))

```

Comment:
Denial is more costly:

To make scope checking more efficient and less costly, a good practice is to invert the condition.

You can also use a method that minimizes the number of iterations required in the collections, making sure that all required scopes are present in the user scopes.

```c#
if (routeAllowedScopes.TrueForAll(userScopes.Contains))
{
    return new OkResponse<bool>(true);
}

return new ErrorResponse<bool>(
    new ScopeNotAuthorizedError($"User scopes: '{string.Join(",", userScopes)}' do not have all allowed scopes: '{string.Join(",", routeAllowedScopes)}'"));
```

Reply:
Right! Thanks! 
It makes sense to invert the condition.